src/prefect/events/cli/automations.py

```diff
@@ -149,7 +150,7 @@ async def inspect(
     if yaml or json:
         if isinstance(automation, list):
             automation = [a.dict(json_compatible=True) for a in automation]
-        elif isinstance(automation, dict):
+        elif isinstance(automation, Automation):
```

Comment:
Also fixes small bug here that I just pushed in like 30 mins ago. I add two regression tests for this below:
`test_inspecting_by_id_in_json`
`test_inspecting_by_id_in_yaml`

Reply:
cc @bunchesofdonald